app/logic/emailHandler.py

```diff
@@ -0,0 +1,89 @@
+from flask import Flask, request, render_template
+import yaml, os
+from flask_mail import Mail, Message
+from app.models.interest import Interest
+from app.models.user import User
+from app.models.eventParticipant import EventParticipant
+from app.models.emailTemplate import EmailTemplate
+from app import app
+import sys
+from pathlib import Path
+
+def load_config(file):
+    """ Loads the 'default' file in the config directory. """
+    with open(file, 'r') as ymlfile:
+        cfg = yaml.load(ymlfile, Loader=yaml.FullLoader)
+    return cfg
+
+
+def getVolunteerEmails(programID = None, eventID = None, emailRecipients = "interested"):
+    """ Gets emails of students interested in a program or participating in an event. """
+    if emailRecipients == 'interested':    #email all students interested in the program
+        volunteersToEmail = User.select().join(Interest).where(Interest.program == programID)
+
+    elif emailRecipients == 'eventParticipant':  #email only people who rsvped
+        volunteersToEmail = User.select().join(EventParticipant).where(EventParticipant.event == eventID)
+
+    else:
+        return None
+    return [user.email for user in volunteersToEmail]
+
+
+class emailHandler():
+    """ A class for email setup and configuring the correct data to send. """
+    def __init__(self, emailInfo):
+        self.default = load_config('app/config/default.yml')
+        app.config.update(
+            MAIL_SERVER=self.default['mail']['server'],
+            MAIL_PORT=self.default['mail']['port'],
+            # MAIL_USERNAME= default['mail']['username'],
+            # MAIL_PASSWORD= default['mail']['password'],
+            REPLY_TO_ADDRESS= self.default['mail']['reply_to_address'],
+            MAIL_USE_TLS=self.default['mail']['tls'],
+            MAIL_USE_SSL=self.default['mail']['ssl'],
+            MAIL_DEFAULT_SENDER=self.default['mail']['default_sender'],
+            MAIL_OVERRIDE_ALL=self.default['mail']['override_addr']
+        )
+
+        self.emailInfo = emailInfo
+        self.mail = Mail(app)
+        self.mail.connect()
+
+    def updateSenderEmail(self):
+        """ Updates who is sending the emails based on the event_list form. """
+        if '@' in self.emailInfo['emailSender']: #if the current user is sending the email

```

Comment:
Note: we can use a mailto link so they send the email from their own mail client

Reply:
This will be put in as a separate issue.